packages/compiler-vapor/src/transforms/transformElement.ts

```diff
@@ -123,14 +123,18 @@ function transformComponentElement(
     }
   }
 
+  const hasVFor = node.props.some(
+    p => p.type === NodeTypes.DIRECTIVE && p.name === 'for',
+  )
+
   context.dynamic.flags |= DynamicFlag.NON_TEMPLATE | DynamicFlag.INSERT
   context.dynamic.operation = {
     type: IRNodeTypes.CREATE_COMPONENT_NODE,
     id: context.reference(),
     tag,
     props: propsResult[0] ? propsResult[1] : [propsResult[1]],
     asset,
-    root: singleRoot,
+    root: singleRoot && !hasVFor,
```

Comment:
```suggestion
    root: singleRoot && !context.inVFor,
```

Reply:
Here you can use `context.inVFor` being 0 to determine whether the current node is in `vFor`

Reply:
@edison1105 Thank you so much. I updated as your suggestion.
Please review my code again.

Reply:
The snapshot is incorrect because `TransformVFor` is after `TransformElement`
I've made minor tweaks, see https://github.com/vuejs/core/pull/13149/commits/a951a739db8425321c3f5f41e671cbe81fbdf86c

Reply:
@edison1105 Thank you for helping me.
If you have more suggestions, please let me know, and I will updatethe  code soon.